arch/arm/core/aarch32/irq_relay.S

```diff
@@ -26,7 +26,10 @@
 
 _ASM_FILE_PROLOGUE
 
-#if defined(CONFIG_SW_VECTOR_RELAY)
+/* When CONFIG_BOOTLOADER_MCUBOOT is set the firmware image
+ * is chain-loadable and does not need to forward interrupts.
+ */
+#if defined(CONFIG_SW_VECTOR_RELAY) && !defined(CONFIG_BOOTLOADER_MCUBOOT)
```

Comment:
I wonder if this would be better encoded in Kconfig.

Reply:
I agree. There could be a concept of VECTOR_RELAY_CLIENT or similar, which just allocates the RAM word for the pointer.

Reply:
But in that case (being a chain-loadable image) why would you enable `SW_VECTOR_RELAY` at all?

Reply:
You wouldn't. You would enable SW_VECTOR_RELAY_CLIENT instead

Reply:
Yeap, @oyvindronningstad is correct here